### PR TITLE
Support only deploying recommender

### PR DESCRIPTION
**Pull request by DebakelOrakel**
## Checklist

- [x] PR contains a single logical change (to build a better changelog).
- [x] Update the documentation.
- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog.
- [ ] Link this PR to related issues or PRs.

### CODE REVIEW

**Review by glrf — CHANGES_REQUESTED** [review]
I think you currently still deploy the admission controller and updater but remove their permissions. Is that intended, sound wrong?

Also please add another `tests/full.yml` or something like that, that has `recommender_only: false` so that changes are recorded in golden tests.

And as a side note. I'm not sure if `recommender_only: true` is an intuitive default, but I don't have strong feelings there and I'm fine with leaving that.

**Comment by DebakelOrakel** [comment]
> And as a side note. I'm not sure if `recommender_only: true` is an intuitive default, but I don't have strong feelings there and I'm fine with leaving that.

Yeah, I was thinking about that myself ...and decided for the "cleaner" implementation codewise (if that makes sense)

File: dependencies/vertical-pod-autoscaler/vpa-v1-crd-gen.yaml [hunk 1]
```diff
@@ -0,0 +1,757 @@
...
+spec:
+  group: autoscaling.k8s.io
+  names:
+    kind: VerticalPodAutoscalerCheckpoint
+    listKind: VerticalPodAutoscalerCheckpointList
+    plural: verticalpodautoscalercheckpoints
+    shortNames:
+    - vpacheckpoint
+    singular: verticalpodautoscalercheckpoint
+  scope: Namespaced
```
Comment (glrf) [inline dependencies/vertical-pod-autoscaler/vpa-v1-crd-gen.yaml:18]: That also shouldn't be committed. I'd suggest you do a rebase and redo that commit.

**Review by glrf — APPROVED** [review]
LGTM, would be nice if you could cleanup the history, but it's also fine if you don't

**Pull request merged**